openmls/src/treesync/mod.rs

```diff
@@ -505,4 +462,13 @@ impl TreeSync {
     pub(crate) fn is_leaf_in_tree(&self, leaf_index: LeafNodeIndex) -> bool {
         is_node_in_tree(leaf_index.into(), self.tree.size())
     }
+
+    /// Return a vector containing all [`HpkePublicKey`]s for which we should
+    /// have the corresponding private keys.
+    pub(crate) fn owned_encryption_keys(&self) -> Vec<EncryptionKey> {
+        self.empty_diff()
```

Comment:
If this weren't on an empty diff (temp value), we could avoid owning the keys and collecting. Not sure if that's easily doable though ...

Reply:
Yeah, the proper solution for this is to have a `Treelike` trait that is implemented by both tree and diff. Then we could more easily share code between the two. I'll file an issue for this so we don't forget. For now creating an empty diff is unfortunately the easiest way.